File: BusinessLogicFSharpDotnetFramework/IndexModule.fs
module BusinessLogicInFSharpProject.IndexModule

type OHLC =
    {
        o: float
        c: float
        h: float
        l: float
    }

// Parse CSV and return OHLC prices
let getOHLC() =
    System.IO.File.ReadAllText($"data.csv")
    |> fun data -> data.Replace(',', '.')
    |> fun data -> data.Replace('\r', ' ')
    |> fun data -> data.Split('\n')
    |> Seq.tail
    |> Seq.map (fun s -> s.Trim())
    |> Seq.map (fun s -> s.Split(';'))
    |> Seq.map (fun s -> 
        { 
            o = float s.[0]
            h = float s.[1]
            l = float s.[2]
            c = float s.[3]
        } : OHLC)
    |> Seq.truncate 20

File: BusinessLogic.ImplementationWithCSharp/IndexModule.cs
using BusinessLogic.Abstraction;
using BusinessLogic.Models;

namespace BusinessLogic.ImplementationWithCSharp
{
    public class IndexModule : IIndexModule
    {
        public IEnumerable<IOHLC> GetOHLC()
        {
            var data = System.IO.File.ReadAllText($"data.csv");
            var dataArray = data.Split("\r\n");

            return dataArray
                .Skip(1)
                .Select(x => x.Split(';'))
                .Select(x => new OHLC
                {
                    O = double.Parse(x[0]),
                    H = double.Parse(x[1]),
                    L = double.Parse(x[2]),
                    C = double.Parse(x[3])
                })
                .Take(20);
        }
    }
}